### PR TITLE
W-7252216 Adding documentation for list membership

### PR DESCRIPTION
**Pull request by gupta298**

### CODE REVIEW

File: docs/kb/api-version-3/export.md [hunk 1]
```diff
@@ -66,12 +66,84 @@ OR
...
+### filter_by_created_at
+
+Retrieves all list membership records with a `created_at` value that is equal or greater than the `created_after` argument and less than or equal to the `created_before` argument.
+
+#### Abilities
+
+| Action           | Requirements  |
+| ---------------- | ------------- |
+| Create export    | Marketing > Segmentation > Lists > View ability `and` Prospects > Visitors > View ability |
+| View export      | Marketing > Segmentation > Lists > View ability `and` Prospects > Visitors > View ability and ability `and` be the same user that created the export |
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
| View export      | Marketing > Segmentation > Lists > View ability `and` Prospects > Visitors > View ability and ability `and` must be the same user that created the export |
```

File: docs/kb/api-version-3/export.md [hunk 2]
```diff
@@ -66,12 +66,84 @@ OR
...
+| View all exports | Admin > Exports > View ability |
+| Query exports    | Admin > Exports > View ability |
+
+
+To create an export with this procedure, the user must have the following:
+
+* “Marketing > Segmentation > Lists > View“ ability AND
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]: Consider standardizing AND & 'and'.

File: docs/kb/api-version-3/export.md [hunk 3]
```diff
@@ -66,12 +66,84 @@ OR
...
 * **created_before**: (Optional) Selects visitor activities that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
 
-__NOTE:__ The range between created_after and created_before cannot exceed 1 year. When created_before is not specified, the current date is used to gauge the interval.
+## List Membership
+
+### filter_by_created_at
+
+Retrieves all list membership records with a `created_at` value that is equal or greater than the `created_after` argument and less than or equal to the `created_before` argument.
+
+#### Abilities
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]: Do the table and the bullet points below contain the same information? Consider cutting one.

File: docs/kb/api-version-3/export.md [hunk 4]
```diff
@@ -66,12 +66,84 @@ OR
...
+* “Prospects > Visitors > View“ ability AND
+* The user must be the same as the user that created the export
+
+OR
+
+* “Admin > Exports > View”
+
+#### Arguments
+
+* **created_after**: Selects list membership that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
* **created_after**: Selects list memberships that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-3/export.md [hunk 5]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **created_before**: (Optional) Selects list membership that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
* **created_before**: (Optional) Selects list memberships that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-3/export.md [hunk 6]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **deleted**: (Optional) Selects list membership based on if it is deleted or not. The value can be `true`, `false`, or `all`.
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
* **deleted**: (Optional) Selects list memberships based on if it is deleted or not. The value can be `true`, `false`, or `all`.
```

Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]: what is "it" referring to here? It seems weird to refer to list membership as deleted. That's not how we typically talk about list membership. Does this refer to prospects being removed from a list? Consider your audience and revise.

File: docs/kb/api-version-3/export.md [hunk 7]
```diff
@@ -66,12 +66,84 @@ OR
...
+
+### filter_by_updated_at
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]: Do the bulleted items and the table give the same information? If so, consider cutting one.

File: docs/kb/api-version-3/export.md [hunk 8]
```diff
@@ -66,12 +66,84 @@ OR
...
+* “Prospects > Visitors > View“ ability AND
+* The user must be the same as the user that created the export
+
+OR
+
+* “Admin > Exports > View”
+
+#### Arguments
+
+* **updated_after**: Selects list membership that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
* **updated_after**: Selects list memberships that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-3/export.md [hunk 9]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **updated_before**: (Optional) Selects list membership that were created before the specified time. This value must be after the value in `updated_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `updated_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
* **updated_before**: (Optional) Selects list memberships that were created before the specified time. This value must be after the value in `updated_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `updated_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-3/export.md [hunk 10]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **deleted**: (Optional) Selects list membership based on if it is deleted or not. The value can be `true`, `false`, or `all`.
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]: Same question as before about what "it" is referring to.

File: docs/kb/api-version-4/export.md [hunk 11]
```diff
@@ -39,11 +39,11 @@ A procedure is a query and execution plan used to retrieve the data. Each object
 
 ## Visitor Activity
 
-## filter_by_created_at
+### filter_by_created_at
 
 Retrieves all visitor activity records with a `created_at` value that is equal or greater than the `created_after` argument and less than or equal to the `created_before` argument.
 
-### Abilities
+#### Abilities
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]: Check your table formatting here, this looks odd.

File: docs/kb/api-version-4/export.md [hunk 12]
```diff
@@ -66,12 +66,84 @@ OR
...
 * **created_before**: (Optional) Selects visitor activities that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
 
-__NOTE:__ The range between created_after and created_before cannot exceed 1 year. When created_before is not specified, the current date is used to gauge the interval.
+## List Membership
+
+### filter_by_created_at
+
+Retrieves all list membership records with a `created_at` value that is equal or greater than the `created_after` argument and less than or equal to the `created_before` argument.
+
+#### Abilities
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]: Do the bulleted list and table give the same information? If so, consider cutting one

File: docs/kb/api-version-4/export.md [hunk 13]
```diff
@@ -66,12 +66,84 @@ OR
...
+* “Prospects > Visitors > View“ ability AND
+* The user must be the same as the user that created the export
+
+OR
+
+* “Admin > Exports > View”
+
+#### Arguments
+
+* **created_after**: Selects list membership that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]:
```suggestion
* **created_after**: Selects list memberships that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-4/export.md [hunk 14]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **created_before**: (Optional) Selects list membership that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]:
```suggestion
* **created_before**: (Optional) Selects list memberships that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-4/export.md [hunk 15]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **deleted**: (Optional) Selects list membership based on if it is deleted or not. The value can be `true`, `false`, or `all`.
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]:
```suggestion
* **deleted**: (Optional) Selects list memberships based on if it is deleted or not. The value can be `true`, `false`, or `all`.
```

File: docs/kb/api-version-3/export.md [hunk 16]
```diff
@@ -39,11 +39,11 @@ A procedure is a query and execution plan used to retrieve the data. Each object
```
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]: General note about the deleted args: 

"Selects list membership based on if it is deleted or not. The value can be `true`, `false`, or `all`"

If you're referring to an object's records here, I'd consider rephrasing all of these to something like "Selects list membership records based on whether they are deleted. The value can be ...."

File: docs/kb/api-version-3/export.md (hunk 5 above)
Comment (humeansacrifice) [inline docs/kb/api-version-3/export.md]:
```suggestion
* **created_before**: (Optional) Selects list membership records that were created before the specified time. This value must be after the value in `created_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `created_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-4/export.md [hunk 17]
```diff
@@ -66,12 +66,84 @@ OR
...
+
+### filter_by_updated_at
+
+Retrieves all list membership records with a `updated_at` value that is equal or greater than the `updated_after` argument and less than or equal to the `updated_before` argument.
+
+#### Abilities
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]: Do the table and the bullet points give the same information? Consider cutting one.

File: docs/kb/api-version-4/export.md [hunk 18]
```diff
@@ -66,12 +66,84 @@ OR
...
+* “Prospects > Visitors > View“ ability AND
+* The user must be the same as the user that created the export
+
+OR
+
+* “Admin > Exports > View”
+
+#### Arguments
+
+* **updated_after**: Selects list membership that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]:
```suggestion
* **updated_after**: Selects list membership records that were created after the specified time. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-4/export.md [hunk 19]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **updated_before**: (Optional) Selects list membership that were created before the specified time. This value must be after the value in `updated_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `updated_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]:
```suggestion
* **updated_before**: (Optional) Selects list membership records that were created before the specified time. This value must be after the value in `updated_after`. If this argument is not specified, then no upper boundary is used in the query, and all data after the `updated_after` is returned. The value can be `today`, `yesterday`, `last_7_days`, `this_month`, `last_month`, or a custom time specified in [GNU Date Input Syntax](http://www.gnu.org/software/tar/manual/html_node/Date-input-formats.html) format.
```

File: docs/kb/api-version-4/export.md [hunk 20]
```diff
@@ -66,12 +66,84 @@ OR
...
+* **deleted**: (Optional) Selects list membership based on if it is deleted or not. The value can be `true`, `false`, or `all`.
```
Comment (humeansacrifice) [inline docs/kb/api-version-4/export.md]:
```suggestion
* **deleted**: (Optional) Selects list membership records based on whether they are deleted. The value can be `true`, `false`, or `all`.
```

**Comment by jpearson-sfdc** [comment]
Before shipping this update to the docs, we need to finish perf testing and enable the feature flag for all accounts. Since this repo is public, we usually don't put pre-production documentation up; we usually fork the repo on the internal instance and then push to production after the feature goes live.

**Pull request closed without merging**